### PR TITLE
Add warning when creating/updating jobs that overlap.

### PR DESCRIPTION
**Pull request by shamrickus**
## What does this PR (Pull Request) do?

- [x] This PR is not related to any Issue

Currently, an invalidation request with _identical_ regex's that overlap in time is able to be created (Currently ATS will ignore the new entry, but this is subject to change). This PR changes it so that warning messages are returned.

## Which Traffic Control components are affected by this PR?

- Traffic Ops
- Traffic Portal

## What is the best way to verify this PR?
Run TO/TP tests
Verify that creating/updating jobs through TO API with conflicting jobs will return relevant warnings.
Verify that creating jobs through TP with conflicting jobs will show the warnings.

## The following criteria are ALL met by this PR

- [x] This PR includes tests OR I have explained why tests are unnecessary
- [x] This PR includes documentation OR I have explained why documentation is unnecessary
- [x] This PR includes an update to CHANGELOG.md OR such an update is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR ensures that database migration sequence is correct OR this PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

File: traffic_portal/app/src/common/service/utils/ServerUtils.js [hunk 1]
```diff
@@ -67,6 +67,24 @@ var ServerUtils = function($window, propertiesModel, userModel) {
 		);
 	};
 
+	// Takes a response object from TO and converts all of its alerts to messages acceptable by the messageModel
+	this.convertResponseToMessages = function(response) {
```
Comment (mitchell852) [inline traffic_portal/app/src/common/service/utils/ServerUtils.js:71]: pretty sure you don't have to do this. messagemodel takes in an array or messages so all you have to do is

```
					messageModel.setMessages(result.alerts, true);
```

or 

```
messageModel.setMessages(result.data.alerts, true);
```

Comment (mitchell852) [reply traffic_portal/app/src/common/service/utils/ServerUtils.js:71]: actually, i guess i'm not sure what you're doing. if the server returns 1 alert, TP will show 1. If it returns 5, TP will show 5...

Comment (mitchell852) [reply traffic_portal/app/src/common/service/utils/ServerUtils.js:71]: it looks like you return 2 alerts, right? one is a success and one is a warning. an alerts array of 2 _should_ work fine...

Comment (shamrickus) [reply traffic_portal/app/src/common/service/utils/ServerUtils.js:71]: You are correct, when I started writing the method I assumed they were differently structured, and I never revisited. It sounds like it would be a better idea to have that method combine all the alerts with the same level instead .

Comment (mitchell852) [reply traffic_portal/app/src/common/service/utils/ServerUtils.js:71]: > It sounds like it would be a better idea to have that method combine all the alerts with the same level instead 

Maybe. Or just leave as it is...

**Comment by mitchell852** [comment]
@shamrickus - you'll need to make a change to FormNewJobController.js as well as the response message is currently hardcoded in there.

You may not have been aware that you can create jobs here in TP as well: https://tp.domain.tld/#!/jobs

**Comment by mitchell852** [comment]
Nit: How about getting these messages in line with each other?

image

I.e.

```
Invalidation request duplicate found for http://foo.bar.com/bingo, start:2020-10-07 21:15:47.444 +0000 UTC end:2020-10-09 21:15:47.444 +0000 UTC
```

```
Invalidation request created for http://foo.bar.com/bingo, start:2020-10-07 21:15:47.444 +0000 UTC end:2020-10-09 21:15:47.444 +0000 UTC
```

File: traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices.go [hunk 2]
```diff
@@ -902,7 +902,7 @@ func updateV30(w http.ResponseWriter, r *http.Request, inf *api.APIInfo, ds *tc.
 		newCDNName, err = getCDNName(*ds.CDNID, tx)
 		newHostName, err = getHostName(ds.Protocol, *ds.Type, *ds.RoutingName, *ds.MatchList, cdnDomain)
 		if err != nil {
-			return nil, http.StatusInternalServerError, nil, errors.New("getting hostname after update: " + err.Error())
+			return nil, http.StatusInternalServerError, nil, errors.New("getting cdnname/hostname after update: " + err.Error())
```
Comment (mitchell852) [inline traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices.go:905]: is this related to this PR?

**Comment by mitchell852** [comment]
weird branch name but no worries just threw me off a bit :)

File: traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices.go (hunk 2 above)
Comment (shamrickus) [reply traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices.go:905]: Yep, I added getCDNName but didn't update the message

**Comment by shamrickus** [comment]
> weird branch name but no worries just threw me off a bit :)

Yeah...I definitely got my wires crossed on this one haha

**Comment by mitchell852** [comment]
```
Invalidation request created for /dup, start:2020-10-08 15:34:03.464 +0000 UTC end 2020-10-09 15:34:03.464 +0000 UTC
```
^^ can you make sure the origin is prepended to the regex in the success message?

**Comment by mitchell852** [comment]
I'm also wondering if the changelog message should be different if it is a "duplicate". here's what it looks like now:

```
Created content invalidation job - ID: 3817 DS: xml-id URL: 'http://origin.tld/dup' Params: 'TTL:24h'
```

maybe something like:

```
Created content invalidation job (duplicate) - ID: 3817 DS: xml-id URL: 'http://origin.tld/dup' Params: 'TTL:24h'
```

**Review by mitchell852 — APPROVED** [review]
works as expected. lgtm.
